### PR TITLE
Feature/gh 5 openstack boot volume

### PR DESCRIPTION
**Pull request by laurentganne**
Port of Yorc Alien4Cloud plugin pull request https://github.com/ystia/yorc-a4c-plugin/pull/143 to alien4cloud-yorc-provider:
added a BootVolume data type, and an optional property boot_volume of this type in yorc.nodes.openstack.Compute.

Fixes issue #5

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -23,7 +23,7 @@
     <yorc.google.types.version>1.0.0</yorc.google.types.version>
     <yorc.hp.types.version>1.0.0</yorc.hp.types.version>
     <yorc.k8s.types.version>2.0.0-SNAPSHOT</yorc.k8s.types.version>
-    <yorc.os.types.version>1.1.0</yorc.os.types.version>
+    <yorc.os.types.version>1.2.0</yorc.os.types.version>
```
Comment (adidanes) [inline pom.xml]: Need to pass in SNAPSHOT os and slurm types

Comment (laurentganne) [reply pom.xml]: ok, merged this branch with develop branch which has already this change after your last push.

**Review by adidanes — APPROVED** [review]
Seems OK To me

File: src/main/resources/openstack/resources/resources.yaml [hunk 2]
```diff
@@ -77,13 +110,18 @@ node_types:
...
-        description: The Openstack VM Image ID either this property or the 'imageName' property is required
+        description: >
+          Openstack VM Image ID, this property is required when 'imageName' is not set and not booting from a volume
         required: false
       imageName:
         type: string
         description: >
-          The Openstack VM Image Name either this property or the 'image' property is required. This property can be used only if the image
-          name is unique.
+          Openstack VM Image Name, this property is required when 'image' is not set and not booting from a volume
```
Comment (loicalbertin) [inline src/main/resources/openstack/resources/resources.yaml]:
```suggestion
          Openstack VM Image Name, this property is required when 'image' is not set and not booting from a volume. This property can be used only if the image name is unique.
```
Still true isn't it?

Comment (laurentganne) [reply src/main/resources/openstack/resources/resources.yaml]: Not described in terraform openstack provider documentation, but yes looking at the implementation, it calls this function https://github.com/gophercloud/gophercloud/blob/master/openstack/compute/v2/images/requests.go#L74 returning an error if more than one image is found with this name.  Will add this in the description.

**Review by loicalbertin — APPROVED** [review]
LGTM :+1: :clap:

**Pull request merged**